Added kana and hanzi to patterns.ts

SkyCoder99 opened the pull request:
Added counter styles for Japanese hiragana and katakana, as well as for the numbers 1-100 in hanzi (Chinese characters). I think there's probably a better way to do the hanzi, either similar to how the roman numerals are done or using built-in numeral parsing, but I'm not sure how.

Here's how hanzi numbers work:
一，二，三，四，五，六，七，八，九  1, 2, 3, 4, 5, 6, 7, 8, 9
十 10; 十一 11; etc.
二十 20; 二十一 21; etc.
一百 100； 一百一 101; etc.
一千 1,000; 一万 10,000

SkyCoder99 committed:
Added kana and hanzi to patterns.ts

Added counter styles for hiragana, katakana, and the numbers 1-100 in hanzi (Chinese characters). I think there's probably a better way to do the hanzi, similar to how the roman numerals are done or if they have underlying code to act as numerals, but I'm not sure how.

SkyCoder99 commented:
...darn, I'm not sure why, but when actually used in Obsidian, it doesn't iterate. (So, ア just gives ア every line instead of ア、イ、ウ,. etc.)

willem-schlieter commented:
Thanks for your contribution!

I think the method for hanzi is fine. No idea how to parse this as `@counter-style` but I think fallback to decimal from 101 on should be fine.

Is there a reason why the Japanese numbers are limited to 51 items?

To make the plugin actually parse the numbers, please add the respective "1"s in the RegEx patterns [in line 12](https://github.com/willem-schlieter/lawlist/pull/11/commits/4e9bd2609f220ed5fb5c6efe2bcfbdb47b7ce9a5#diff-8d1d42c2c6eb270a557ae0ebb4719b7ea654658a94c2f0df2ed03cad4107efd9R12) and [124](https://github.com/willem-schlieter/lawlist/pull/11/commits/4e9bd2609f220ed5fb5c6efe2bcfbdb47b7ce9a5#diff-8d1d42c2c6eb270a557ae0ebb4719b7ea654658a94c2f0df2ed03cad4107efd9R124) so that they are recognised as number characters. (Currently, they aren't so the plugin handles them the same way as dots, hyphens etc.) As far as I see it, that should fix it.

SkyCoder99 commented:
Cool!

So the Japanese ones are letters rather than numbers, is why they're
limited to 51.

I'll go do those fixes.

On Thu, Jun 19, 2025, 5:51 AM Willem Rubeta Schlieter <
***@***.***> wrote:

> *willem-schlieter* left a comment (willem-schlieter/lawlist#11)
> <https://github.com/willem-schlieter/lawlist/pull/11#issuecomment-2987811457>
>
> Thanks for your contribution!
>
> I think the method for hanzi is fine. No idea how to parse this as
> @counter-style but I think fallback to decimal from 101 on should be fine.
>
> Is there a reason why the Japanese numbers are limited to 51 items?
>
> To make the plugin actually parse the numbers, please add the respective
> "1"s in the RegEx patterns in line 12
> <https://github.com/willem-schlieter/lawlist/pull/11/commits/4e9bd2609f220ed5fb5c6efe2bcfbdb47b7ce9a5#diff-8d1d42c2c6eb270a557ae0ebb4719b7ea654658a94c2f0df2ed03cad4107efd9R12>
> and 124
> <https://github.com/willem-schlieter/lawlist/pull/11/commits/4e9bd2609f220ed5fb5c6efe2bcfbdb47b7ce9a5#diff-8d1d42c2c6eb270a557ae0ebb4719b7ea654658a94c2f0df2ed03cad4107efd9R124>
> so that they are recognised as number characters. (Currently, they aren't
> so the plugin handles them the same way as dots, hyphens etc.) As far as I
> see it, that should fix it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/willem-schlieter/lawlist/pull/11#issuecomment-2987811457>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BDFRRZF5TXNDIJOFTSCYMML3EKP57AVCNFSM6AAAAAB7SHPAWCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSOBXHAYTCNBVG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

willem-schlieter commented:
I see! So at the moment, `system: fixed` makes the rule fall back to decimal numbers after the lookup ends, i.e. from the 52th list item on, which is fine. Did you choose this in favour of `system: cyclic` (A, B, C … Y, Z, A, B) or `system: alphabetic` ("A, B … Y, Z, AA, AB", which I used for latin letters)? I've got no clue what would make sense in Japanese, just as an idea...

SkyCoder99 committed:
Update patterns.ts

changed hiragana and katakana to system: alphabetic
added to the RegEx patterns in lines 12 and 124
